### PR TITLE
zsh-abbr: update source path

### PR DESCRIPTION
**Pull request by ilaumjd**
### Description

Path changes on nixpkgs: https://github.com/NixOS/nixpkgs/commit/d90af1448cfa1d7a013c80d435878023f92471bf

### Checklist

- [ ] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all` or `nix develop --ignore-environment .#all` using Flakes.

- [ ] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://nix-community.github.io/home-manager/#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [ ] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

### CODE REVIEW

**Comment by champloo** [comment]
Noting that this PR change is the same as another #5963 authored by @Icy-Thought  who made the related changed in nixpkgs https://github.com/NixOS/nixpkgs/pull/348162

**Comment by Icy-Thought** [comment]
Thanks for mentioning that @champloo & pinging me! I can close my PR if y'all want to proceed with this one, as long as one of them gets merged!

File: modules/programs/zsh/zsh-abbr.nix [hunk 1]
```diff
@@ -28,7 +28,7 @@ in {
     programs.zsh.plugins = [{
       name = "zsh-abbr";
       src = pkgs.zsh-abbr;
-      file = "/share/zsh/zsh-abbr/abbr.plugin.zsh";
```
[the remark this answers is not in the document]
Comment (teto) [reply modules/programs/zsh/zsh-abbr.nix:31]: it's not done inthe referenced PR https://github.com/nix-community/home-manager/pull/5963/files

[the remark this answers is not in the document]
Comment (champloo) [reply modules/programs/zsh/zsh-abbr.nix:31]: Yup I noticed this too. Pretty sure this is to fix the current behaviour where  the path written out to '.zshrc` contains double slashes `$HOME/.zsh/plugins/zsh-abbr//share/zsh/zsh-abbr/abbr.plugin.zsh`. @ilaumjd should be able to confirm

**Comment by ilaumjd** [comment]
@champloo @Icy-Thought Oh I'm sorry I didn't aware about the other PR, should be doing more checking later. Anything is fine for me as long one of it got merged.

File: modules/programs/zsh/zsh-abbr.nix (hunk 1 above)
[the remark this answers is not in the document]
Comment (ilaumjd) [reply modules/programs/zsh/zsh-abbr.nix:31]: Yes, just noticed that the path was having double slashes before.

**Pull request merged**